misk-feature/src/main/kotlin/misk/feature/RealAppScopedDynamicConfigResolver.kt

```diff
@@ -0,0 +1,31 @@
+package misk.feature
+
+import misk.logging.getLogger
+
+internal class RealAppScopedDynamicConfigResolver<T : ValidatableConfig<T>> constructor(
+  private val appName: String,
+  private val configName: String,
+  private val dynamicConfig: DynamicConfig,
+  private val failureModeDefault: T
+) : AppScopedDynamicConfigResolver<T> {
+  private val logger = getLogger<AppScopedDynamicConfigResolver<T>>()
+
+  override fun resolveConfig(): T {
+    require(!configName.startsWith(appName)) {
+      "Redundant '$appName' prefix for ConsumerConfigResolver managed app config '$configName'"
+    }
+
+    val featureName = listOf(appName, configName).joinToString("-")
```

Comment:
it's not clear that the feature must be named like this

Reply:
why enforce this anyways?

Reply:
This is probably overly prescriptive, but I wanted to account for the not-yet-enforced convention of scoping feature flags to a service by prefixing them with the app name.

I want to reduce the chances of collisions between features registered for misk service deployments in an environment sharing one namespace for all their features (as we do with LaunchDarkly)

Reply:
What if I just require this by validation instead of constructing it myself here? 
I.e., I'll just start with `require(configName.startsWith(appName))` instead of `require(!configName.startsWith(appName))`

Then, implementors will be on the same page and the code that registers these config names will both have to start with the app name *and* it will appear in code as it is registered in the feature flag service (e.g., LD)